micrometer-support/src/main/java/io/javaoperatorsdk/operator/monitoring/micrometer/MicrometerMetrics.java

```diff
@@ -31,9 +31,17 @@ public class MicrometerMetrics implements Metrics {
   private static final String RECONCILIATIONS_QUEUE_SIZE = PREFIX + RECONCILIATIONS + "queue.size.";
   private final MeterRegistry registry;
   private final Map<String, AtomicInteger> gauges = new ConcurrentHashMap<>();
+  private final Map<ResourceID, Set<Meter.Id>> metersPerResource = new ConcurrentHashMap<>();
+  private final ScheduledExecutorService metersCleaner = Executors.newScheduledThreadPool(10);
```

Comment:
Since the clean is a short running task (I assume), wouldn't'  be better to use a Timer?

Reply:
Because it's usually better to use `ScheduledExecutorService` than `Timer`. See https://stackoverflow.com/a/409993.

Reply:
Ok, then the number of Threads should be configurable, and we should use 1 as default.

Reply:
Actually the whole ScheduledExecutorService should be configurable.

Reply:
Both delay and thread numbers are configurable. What other aspects would need configuration?

Reply:
For example bursting the threads (like the cached executor), but I think it fine for now now